### ISSUE TITLE
Current status of `alert_email` `UserOption`?

### ISSUE THREAD

**Issue by mattrobenolt** [getsentry/sentry#4490]
As of right now, we have this `alert_email` setting that is still being referenced in code, causing email to be delivered if this value is set. But this value isn't exposed at all in the UI.

So tracing back, what happens is, someone set this value prior to https://github.com/getsentry/sentry/pull/4130 being a thing. They now have an email address set in `alert_email`. This PR went out, causing this value to become invisible to a user, but emails may still be sent here.

The "fix" at the moment, is to go to the notification settings page and simply click "Save Changes" without any changes, which will cause the value to get saved back into the database as an empty string, causing the value to get bypassed for email delivery.

From my understanding, this value is only affecting reports delivery, or rather, any email sent that isn't bound by a project level override. I'm not sure if there are other places, but it seems that the weekly reports are the only case.

What do we do with this current value? Should we just 100% remove this from code and pretend it never existed? Should we introduce a new value specifically for reports or "organization-level notifications"?

Basically, right now, we have two types of situations that aren't great. a) user who previously set this value and expects it to work correctly, so if we change it on them by removing the field, we'd change the behavior for them. b) users who previously had this set, and don't want it set anymore, but can't find this in the UI to change it and are confused why they're getting email delivered here.

I'm happy to implement whichever case here, it's just unclear to me what the intended behavior should be.

@dcramer @kjlundsgaard

**Comment by mitsuhiko** [comment]
Do we store the new selector stuff elsewhere?

**Comment by mattrobenolt** [comment]
No.

**Comment by kjlundsgaard** [comment]
Something to note is that if the user did have the `alert_email` set then it should appear as the notification email in the dropdown in notification settings. The idea was to protect legacy data but only allow users to choose from new verified emails for notifications moving forward (see #4029). Users who have previously had `alert_email` set but no longer want email sent there may update the email notification preferences in the dropdown.

I agree that it could be confusing for users and am open to a new solution for this.

**Comment by BYK** [comment]
Closing this issue due to staleness. Feel free to comment here if you think we should still work on this.
